ocaml/stdlib/mutex.mli

```diff
@@ -58,3 +58,14 @@ val unlock : t -> unit
 
    @before 4.12 {!Sys_error} was not raised when unlocking an unlocked mutex
    or when unlocking a mutex from a different thread. *)
+
+val protect : t -> (unit -> 'a) -> 'a
+(** [protect mutex f] runs [f()] in a critical section where [mutex]
+    is locked (using {!lock}); it then takes care of releasing [mutex],
+    whether [f()] returned a value or raised an exception.
+
+    The unlocking operation is guaranteed to always takes place,
+    even in the event an asynchronous exception (e.g. {!Sys.Break}) is raised
```

Comment:
(Is the `Sys.Break` part still true in our fork?)

Reply:
Good catch, this probably needs to use Sys.with_async_exns.  I will fix this as part of the async-exns work.

Reply:
Have added a CR